explorer/Explorer.tsx

```diff
@@ -341,130 +343,275 @@ export class Explorer
                 dimensions.push({
-                    variableId: sizeIndicatorId,
-                    property: DimensionProperty.size,
+                    variableId: indicatorId,
+                    property: DimensionProperty.table, // no specific dimension
                 })
-            }
-            config.dimensions = dimensions
+            })
         }
 
+        config.dimensions = dimensions
+
         grapher.setAuthoredVersion(config)
         grapher.reset()
-        grapher.yAxis.canChangeScaleType = yScaleToggle
-        grapher.yAxis.min = yAxisMin
-        if (facetYDomain) {
-            grapher.yAxis.facetDomain = facetYDomain
-        }
-        if (relatedQuestionText && relatedQuestionUrl) {
-            grapher.relatedQuestions = [
-                { text: relatedQuestionText, url: relatedQuestionUrl },
-            ]
-        }
-        if (mapTargetTime) {
-            grapher.map.time = mapTargetTime
-        }
+        this.updateGrapherFromExplorerCommon()
         grapher.updateFromObject(config)
+        await grapher.downloadLegacyDataFromOwidVariableIds()
+
+        let grapherTable = grapher.inputTable
+
+        // update column definitions with manually provided properties
+        grapherTable = grapherTable.updateDefs((def: OwidColumnDef) => {
+            const manuallyProvidedDef =
+                this.columnDefsNotLinkedToTableByIdOrSlug[def.slug] ?? {}
+            const mergedDef = { ...def, ...manuallyProvidedDef }
+
+            // update display properties
+            mergedDef.display = mergedDef.display ?? {}
+            if (manuallyProvidedDef.name)
+                mergedDef.display.name = manuallyProvidedDef.name
+            if (manuallyProvidedDef.unit)
+                mergedDef.display.unit = manuallyProvidedDef.unit
+            if (manuallyProvidedDef.shortUnit)
+                mergedDef.display.shortUnit = manuallyProvidedDef.shortUnit
+
+            return mergedDef
+        })
 
-        if (
-            creationMode ===
-            ExplorerChartCreationMode.FromExplorerTableColumnSlugs
-        ) {
-            // Clear any error messages, they are likely to be related to dataset loading.
-            this.grapher?.clearErrors()
-            // Set a table immediately. A BlankTable shows a loading animation.
-            this.setGrapherTable(
-                BlankOwidTable(tableSlug, `Loading table '${tableSlug}'`)
+        // add derived columns to grapher table
+        if (uniqueSlugsInGrapherRow.length) {
+            const allColumnSlugs = uniq(
+                uniqueSlugsInGrapherRow.flatMap((slug) => [
+                    ...this.getBaseColumnsForColumnWithTransform(slug),
+                    slug,
+                ])
             )
-            this.futureGrapherTable.set(this.tableLoader.get(tableSlug))
+            const existentColumnSlugs = grapherTable.columnSlugs
+            const requiredColumnSlugs = allColumnSlugs.filter(
```

Comment:
Maybe using the word `outstanding` makes it clearer what's going on here
```suggestion
            const outstandingColumnSlugs = allColumnSlugs.filter(
```